services/frontend-service/src/ui/components/EnvironmentConfigDialog/EnvironmentConfigDialog.tsx

```diff
@@ -0,0 +1,73 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+import React, { useCallback, useEffect, useState } from 'react';
+import { Button } from '../button';
+import { Close } from '../../../images';
+import { PlainDialog } from '../dialog/ConfirmationDialog';
+import { useSearchParams } from 'react-router-dom';
+import { getOpenEnvironmentConfigDialog, setOpenEnvironmentConfigDialog } from '../../utils/Links';
+import { useApi } from '../../utils/GrpcApi';
+
+export type EnvironmentConfigDialogProps = {
+    environmentName: string;
+};
+
+export const EnvironmentConfigDialog: React.FC<EnvironmentConfigDialogProps> = (props) => {
+    const environmentName = props.environmentName;
+    const [params, setParams] = useSearchParams();
+    const api = useApi;
+    const isOpen = useCallback((): boolean => getOpenEnvironmentConfigDialog(params).length > 0, [params]);
+    const onClose = useCallback((): void => {
+        setOpenEnvironmentConfigDialog(params, '');
+        setParams(params);
+    }, [params, setParams]);
+    const [config, setConfig] = useState('');
+    useEffect(() => {
+        if (getOpenEnvironmentConfigDialog(params) !== environmentName) {
+            setConfig('loading ...'); // we invisible, so prefill with "loading ..." until the data is here.
+            return;
+        }
+        const result = api.environmentService().GetEnvironmentConfig({ environment: environmentName });
+        result.then((res) => {
+            const pretty = JSON.stringify(res, null, ' ');
+            setConfig(pretty);
+        });
+        result.catch((e) => {
+            // eslint-disable-next-line no-console
+            console.log('error while loading environment config: ' + e);
```

Comment:
For consistency, please extract this to store.tsx where most of our calls are. See `getSummary` for example.

Reply:
But ... that seems wrong. I understand `store.tsx` should contain convenience functions to interface with the backend/api. But here that would only be the `GetEnvironmentConfig` call, while the rest is tightly coupled with the dialog component. Also note that there is one `config, setConfig` state per dialog component.

`getSummary` IMHO is a great example how not to do it TBH:
* it uses a global `useSummary, updateSummary` state, while something component local would be better
* it globally modifies the UI (e.g. the snackbar on error) => shouldnt that be the components call on how to handle?

Reply:
ok, I didn't really explain what my thought behind this was.

Having all grpc calls in one place makes it much easier to make all calls behave consistently. E.g. every call needs a spinner and the state handling that comes with. That's missing from your PR. It would have been no effort, if we would have started by copying `getSummary`.

> it also globally modifies the UI (e.g. the snackbar on error) => shouldnt that be the components call on how to handle?
That's fair.
We should display a human readable error. Snackbar is fine for that, but we can also display an error within the component.
I don't see either of these solutions in this PR though.

When these 2 things are solved, I don't care so much about which file it's in. Personally I would still extract it into a function.

And we should work on some development guidelines together!